packages/source-fetcher/lib/etherscan.ts

```diff
@@ -104,17 +116,45 @@ const EtherscanFetcher: FetcherConstructor = class EtherscanFetcher
   }
 
   private determineUrl() {
-    switch (this.networkName) {
-      case "arbitrum":
-        return "https://api.arbiscan.io/api";
-      case "polygon":
-        return "https://api.polygonscan.com/api";
-      case "mainnet":
-        return "https://api.etherscan.io/api";
-      default:
-        return `https://api-${this.networkName}.etherscan.io/api`;
+    const scanners: { [network: string]: string } = {
+      //etherscan.io is treated separately
+      polygon: "polygonscan.com",
+      arbitrum: "arbiscan.io",
+      binance: "bscscan.com",
+      fantom: "ftmscan.com",
+      //we don't support avalanche's snowtrace.io
+      heco: "hecoinfo.com"
+      //moonscan.io is treated separately
+    };
+    const [part1, part2] = this.networkName.split("-");
```

Comment:
Well, these network names are our own assignations; these aren't standardized names, they're just the names we've assigned them.  So we can alter all this as necessary.  My intent wasn't that this code would work for all future possible networks we might add; indeed, `moonbase-alpha` is already an exception to to this convention (because I found applying that convention to the moon networks to be unworkable, as I mentioned).  My intent was that it would work for these ones, and that in adding future ones, we would alter this as appropriate.

If you're really wary of having to perform such updates, we could use a lookup table instead?  I mean, that would obviously also require updates, but more transpararent ones.  I just wanted to factor out some regularities. :P  But I'll admit that does perhaps require more thought for future maintainers...

Reply:
I'm ok with this being future work if you want to get this in for now.

Reply:
Well I don't think that was the question, I think the question was how to process them into URLs -- lookup table, or my quirky logic here? :)

Reply:
No preference, just came here to state the unnecessary thing that I do have an opinion on :)